### ISSUE TITLE
Feature Request: ProjectManager offers option to do multithreaded builds

### ISSUE THREAD

**Issue by lmbr-pip** [o3de/o3de#15424]
**Is your feature request related to a problem? Please describe.**
O3DE supports multithreaded building but project manager does not offer any config over this.

This means on powerful machines its really slow

```
cmake --build E:/dev/o3de/o3de-netsoaktest/build/windows --config profile --target NetSoakTest.GameLauncher NetSoakTest.ServerLauncher NetSoakTest.UnifiedLauncher Editor
```

Note: No use of `--m`` options: https://www.o3de.org/docs/user-guide/build/configure-and-build/ 

"-m : A recommended build tool optimization. It tells the Microsoft compiler (MSVC) to use multiple threads during compilation to speed up build times."

**Describe the solution you'd like**
Exposes options to use multithreaded builds or control number of cores used for builds

**Describe alternatives you've considered**

**Additional context**

**Comment by lumberyard-employee-dm** [comment]
I think it would be good to add additional user defined cmake arguments to the Project Manager build commands here: https://github.com/o3de/o3de/blob/development/Code/Tools/ProjectManager/Platform/Windows/ProjectBuilderWorker_windows.cpp#L27-L38

That way options such as `-m` or any other build tool specific arguments can be supplied by the user

**Comment by AMZN-alexpete** [comment]
I agree that presets would be great! That way if the user wanted to, they could even make their own preset to share with their team and we can provide common presets like you listed that don't require the user to know all the build tool arguments for common types of builds.
